client/handle_http.go

```diff
@@ -2122,31 +2126,52 @@ func runPut(request *http.Request, responseChan chan<- *http.Response, errorChan
 }
 
 // Walk a remote directory in a WebDAV server, emitting the files discovered
-func (te *TransferEngine) walkDirDownload(job *clientTransferJob, transfers []transferAttemptDetails, files chan *clientTransferFile, url *url.URL) error {
-	// Create the client to walk the filesystem
-	rootUrl := *url
-	if job.job.namespace.DirListHost != "" {
-		// Parse the dir list host
-		dirListURL, err := url.Parse(job.job.namespace.DirListHost)
+func (te *TransferEngine) walkDirDownload(job *clientTransferJob, transfers []transferAttemptDetails, files chan *clientTransferFile, remoteURL *url.URL) error {
+	// Query the director a PROPFIND to see if we can get our directory listing
+	resp, err := queryDirector("PROPFIND", remoteURL.Path, job.job.directorUrl+"/api/v1.0/director/origin")
+	if err != nil {
+		return err
+	}
+
+	collectionsUrl := *remoteURL
+	// If we get a "method not allowed", we go back to the old method
```

Comment:
For someone who doesn't already know what this code does, this comment won't be very elucidating. How about something like "If the director responds with 405 (method not allowed), we're working with an old Director. In that event, we try to fallback and use the deprecated dirlisthost"